conda-store-server/conda_store_server/server/app.py

```diff
@@ -340,16 +349,53 @@ async def favicon():
                 name="static-storage",
             )
 
+        @app.on_event("startup")
+        async def startup_event():
+            import signal
+            import time
+
+            from conda_store_server import orm
+
+            # Makes the loop below exit on Ctrl-C
+            signal.signal(signal.SIGINT, sys.exit)
+
+            # Colors
+            green = "\x1b[32m"
+            red = "\x1b[31m"
+            reset = "\x1b[0m"
+
+            # Waits in a loop for worker to become initialized
+            while True:
+                with self.conda_store.session_factory() as db:
+                    q = db.query(orm.Worker).first()
+                    if q is not None and q.initialized:
+                        logger.info(f"{green}" "Worker initialized" f"{reset}")
+                        break
+
+                time.sleep(5)
+                logger.critical(
+                    f"{red}"
+                    "Waiting for worker... "
+                    "Use --standalone if running outside of docker"
+                    f"{reset}"
+                )
+
         return app
 
     def start(self):
         fastapi_app = self.init_fastapi_app()
 
+        from conda_store_server import orm
+
         with self.conda_store.session_factory() as db:
             self.conda_store.ensure_settings(db)
             self.conda_store.ensure_namespace(db)
             self.conda_store.ensure_conda_channels(db)
 
+            # Needs to delete this entry to check if the worker is available
+            db.query(orm.Worker).delete()
+            db.commit()
```

Comment:
This needs to happen on the server side before the worker is started to avoid a race condition. Moreover, it needs to happen on the server side because it will work in case the server is terminated and there are existing entries in the DB from previous runs.

Reply:
I need to add this as a comment in the code